src/backends/backend/triton_model_instance.cc

```diff
@@ -80,25 +86,59 @@ TritonModelInstance::CreateInstances(
                                     : group.name()};
       const bool passive = group.passive();
       if (group.kind() == inference::ModelInstanceGroup::KIND_CPU) {
+        const std::string policy_name =
+            group.host_policy().empty() ? "cpu" : group.host_policy();
+        const HostPolicyCmdlineConfig* host_policy;
+        const auto policy_it = host_policy_map.find(policy_name);
+        if (policy_it != host_policy_map.end()) {
+          host_policy = &policy_it->second;
+        } else {
+          host_policy = &empty_host_policy;
+        }
+        RETURN_IF_ERROR(SetNumaConfigOnThread(*host_policy));
         RETURN_IF_ERROR(CreateInstance(
             model, instance_name, c, TRITONSERVER_INSTANCEGROUPKIND_CPU,
-            0 /* device_id */, profile_names, passive));
+            0 /* device_id */, profile_names, passive, policy_name,
+            *host_policy));
       } else if (group.kind() == inference::ModelInstanceGroup::KIND_GPU) {
         for (const int32_t device_id : group.gpus()) {
+          const std::string policy_name =
+              group.host_policy().empty() ? ("gpu_" + std::to_string(device_id))
+                                          : group.host_policy();
+          const HostPolicyCmdlineConfig* host_policy;
+          const auto policy_it = host_policy_map.find(policy_name);
+          if (policy_it != host_policy_map.end()) {
+            host_policy = &policy_it->second;
+          } else {
+            host_policy = &empty_host_policy;
+          }
+          RETURN_IF_ERROR(SetNumaConfigOnThread(*host_policy));
           RETURN_IF_ERROR(CreateInstance(
               model, instance_name, c, TRITONSERVER_INSTANCEGROUPKIND_GPU,
-              device_id, profile_names, passive));
+              device_id, profile_names, passive, policy_name, *host_policy));
         }
       } else if (group.kind() == inference::ModelInstanceGroup::KIND_MODEL) {
+        const std::string policy_name =
+            group.host_policy().empty() ? "model" : group.host_policy();
+        const HostPolicyCmdlineConfig* host_policy;
+        const auto policy_it = host_policy_map.find(policy_name);
+        if (policy_it != host_policy_map.end()) {
+          host_policy = &policy_it->second;
+        } else {
+          host_policy = &empty_host_policy;
+        }
+        RETURN_IF_ERROR(SetNumaConfigOnThread(*host_policy));
         RETURN_IF_ERROR(CreateInstance(
             model, instance_name, c, TRITONSERVER_INSTANCEGROUPKIND_MODEL,
-            0 /* device_id */, profile_names, passive));
+            0 /* device_id */, profile_names, passive, policy_name,
+            *host_policy));
       } else {
         return Status(
             Status::Code::INVALID_ARG,
             std::string("instance_group kind ") +
                 ModelInstanceGroup_Kind_Name(group.kind()) + " not supported");
       }
+      RETURN_IF_ERROR(ResetNumaMemoryPolicy());
```

Comment:
I couldn't find a proper way to reset the thread affinity, the way that the custom harness handle it is to launch a separate thread, perform the setup from within the thread. We can adopt the same but it may not be necessary as the model manager actually spawn new thread to create a model.